apps/homescreen/js/app.js

```diff
@@ -540,6 +577,12 @@ const SETTINGS_VERSION = 0;
 
         this.pageHeight = pageHeight;
         this.pendingGridHeight = gridHeight;
+
+        if (!this.visualLoadComplete &&
+            Math.floor(visibleChildren / (this.small ? 4 : 3)) * iconHeight >=
+              scrollHeight) {
+          this.onVisualLoad();
+        }
```

Comment:
Weird, so onVisualLoad is getting called at the right time, but for some reason, you still see the icons load in - It could be platform not loading the image in time... There's code in gaia-site-icon that's meant to make the image fade in on load, but it seems that the image isn't actually ready to display when receiving the onload signal a lot of the time...